### PR TITLE
add boxes-to-polygons transform

### PR DESCRIPTION
**Pull request by jihyeonyi**
### Summary
Add bbox to polygon transform.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: src/datumaro/plugins/transforms.py [hunk 1]
```diff
@@ -266,6 +266,31 @@ def convert_bbox(bbox, img_h, img_w):
         )
 
 
+class BoxesToPolygons(ItemTransform, CliPlugin):
+    def transform_item(self, item):
+        annotations = []
+        for ann in item.annotations:
+            if ann.type == AnnotationType.bbox:
+                if not isinstance(item.media, Image):
+                    raise Exception("Image info is required for this transform")
```
Comment (vinnamkim) [inline src/datumaro/plugins/transforms.py]: Is it still required?

File: src/datumaro/plugins/transforms.py [hunk 2]
```diff
@@ -266,6 +266,31 @@ def convert_bbox(bbox, img_h, img_w):
...
+                annotations.append(self.convert_bbox(ann))
+            else:
+                annotations.append(ann)
```
Comment (vinnamkim) [inline src/datumaro/plugins/transforms.py]:
```suggestion
        annotations = [
             self.convert_bbox(ann) if ann.type == AnnotationType.bbox else ann
             for ann in item.annotations
        ]
```

Comment (jihyeonyi) [reply src/datumaro/plugins/transforms.py]: Oh, you're right. We don't need (w,h) of image for conversion, so, media type condition isn't needed.

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Pull request merged**